python/tank/platform/engine.py

```diff
@@ -798,6 +801,54 @@ def show_modal(self, title, bundle, widget_class, *args, **kwargs):
         # lastly, return the instantiated widget
         return (status, widget)
     
+    def _resolve_sg_stylesheet_tokens(self, style_sheet):
+        """
+        Given a string containing a qt style sheet,
+        perform replacements of key toolkit tokens.
+        
+        For example, "{{SG_HIGHLIGHT_COLOR}}" is converted to "#30A7E3"
+        
+        :param style_sheet: Stylesheet string to process
+        :returns: Stylesheet string with replacements applied
+        """
+        processed_style_sheet = style_sheet
+        for (token, value) in constants.QSS_REPLACEMENTS.iteritems():
+            processed_style_sheet = processed_style_sheet.replace("{{%s}}" % token, value)
+        return processed_style_sheet
+    
+    def _apply_external_styleshet(self, bundle, widget):
+        """
+        Apply an std external stylesheet, associated with a bundle, to a widget.
+        
+        This will check if a standard style.css file exists in the
+        app/engine/framework root location on disk and if so load it from 
+        disk and apply to the given widget. The style sheet is cascading, meaning 
+        that it will affect all children of the given widget. Typically this is used
+        at window creation in order to allow newly created dialogs to apply app specific
+        styles easily.
+        
+        :param bundle: app/engine/framework instance to load style sheet from
+        :param widget: widget to apply stylesheet to 
+        """
+        qss_file = os.path.join(bundle.disk_location, constants.BUNDLE_STYLESHEET_FILE)
```

Comment:
I think it's probably too much at this point! At some point later, maybe we could introduce it if the need arises? In that case maybe you could have a simple override behaviour too?

```
app_root
  |- style.qss        # app level style sheet, gets applied to all UIs
  |- mainwindow.qss   # gets applied to the mainwindow dialog
  \- save_as.qss       # gets applied to the save as dialog
```